src/upload-tests.proj

```diff
@@ -77,6 +77,13 @@
     <HelixArchLabel>$(ArchGroup)</HelixArchLabel>
     <HelixConfigLabel>$(ConfigurationGroup)</HelixConfigLabel>
     <MaxRetryCount>1</MaxRetryCount>
+    <!-- Unique container name per build and configuration, e.g., corefx-beta-12345-01-windowsnt-x64-netcoreapp-master -->
+    <!-- Azure Storage containers can only contain [a-z, 0-9, -] in their names and $(OSGroup) is "Windows_NT" on Windows... -->
+    <ProcessedOSGroup>$([System.Text.RegularExpressions.Regex]::Replace('$(OSGroup.ToLower())', "[^A-Za-z0-9]+", ""))</ProcessedOSGroup>
```

Comment:
Given you are using the same processing for all I'd suggest avoiding all these extra properties and instead just create a `RawContainerName` and then process it all at once into `ContainerName`. That also ensures we catch any additional properties we might add in the future.